Add extra gRPC features to .NET 7 release notes (#27454)
dotnet/AspNetCore.Docs: aspnetcore/release-notes/aspnetcore-7.0.md

For more information, see [gRPC JSON transcoding in ASP.NET Core gRPC apps](xref:grpc/json-transcoding?view=aspnetcore-7.0) and <xref:grpc/json-transcoding-openapi>.

### gRPC health checks in ASP.NET Core

The [gRPC health checking protocol](https://github.com/grpc/grpc/blob/master/doc/health-checking.md) is a standard for reporting the health of gRPC server apps. An app exposes health checks as a gRPC service. They are typically used with an external monitoring service to check the status of an app.

gRPC ASP.NET Core has added built-in support for gRPC health checks with the [`Grpc.AspNetCore.HealthChecks`](https://www.nuget.org/packages/Grpc.AspNetCore.HealthChecks) package. Results from [.NET health checks](xref:host-and-deploy/health-checks) are reported to callers.

For more information, see <xref:grpc/health-checks>.

### Improved call credentials support

Call credentials are the recommended way to configure a gRPC client to send an auth token to the server. gRPC clients support two new features to make call credentials easier to use:

> * Support for call credentials in a local development environment that doesn't have TLS configured: Previously, a gRPC call only sent call credentials if the connection was secured with TLS. A new setting on `GrpcChannelOptions`, called `UnsafeUseInsecureChannelCallCredentials`, allows this behavior to be customized.

Author: ```suggestion
* Support for call credentials with plaintext connections. Previously, a gRPC call only sent call credentials if the connection was secured with TLS. A new setting on `GrpcChannelOptions`, called `UnsafeUseInsecureChannelCallCredentials`, allows this behavior to be customized. There are security implications to not securing a connection with TLS.
```

> * Previously, a gRPC call only sent call credentials if the connection was secured with TLS. A new setting on `GrpcChannelOptions`, called `UnsafeUseInsecureChannelCallCredentials`, allows this behavior to be customized. A scenario where this is useful is using call credentials in a local development that doesn't have TLS configured.

Reviewer: ```suggestion
* Previously, a gRPC call only sent call credentials if the connection was secured with TLS. A new setting on `GrpcChannelOptions`, called `UnsafeUseInsecureChannelCallCredentials`, allows this behavior to be customized. A scenario where this is useful is using call credentials in a local development environment that doesn't have TLS configured.
```

Was `environment` supposed to be in this sentence?

> * Previously, a gRPC call only sent call credentials if the connection was secured with TLS. A new setting on `GrpcChannelOptions`, called `UnsafeUseInsecureChannelCallCredentials`, allows this behavior to be customized. A scenario where this is useful is using call credentials in a local development that doesn't have TLS configured.

Reviewer 2: ```suggestion
* Support for call credentials in a local development environment that doesn't have TLS configured: Previously, a gRPC call only sent call credentials if the connection was secured with TLS. A new setting on `GrpcChannelOptions`, called `UnsafeUseInsecureChannelCallCredentials`, allows this behavior to be customized.
```

Reviewer 2: Minor thing, thought it might be better to lead with the feature description summary.   Added the previous "environment" suggestion so it did not get lost in the shuffle.
